pkg/tls/ca.go

```diff
@@ -262,3 +263,40 @@ func (v *Validity) Window(t time.Time) (time.Time, time.Time) {
 	end := t.Add(life).Add(skew)
 	return start, end
 }
+
+// ParseRootCA parses the provided CA cert and private key. If the cert and key
+// matches, it returns a CA object. If they don't match, ParseRootCA panic. See
+// validCredOrPanic() in pkg/tls/cred.go
+//
+// The Validity field of the returned CA is left empty as the case in
+// GenerateRootCAWithDefaults().
+//
+// Both the CA cert and private key must be PEM-encoded. Otherwise, an error is
+// returned.
+func ParseRootCA(rootCAPEM, keyPEM []byte) (*CA, error) {
+	rootCADecoded, _ := pem.Decode(rootCAPEM)
+	if rootCADecoded == nil {
+		return nil, fmt.Errorf("root CA must be PEM-encoded")
+	}
+
+	rootCA, err := x509.ParseCertificate(rootCADecoded.Bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	keyDecoded, _ := pem.Decode(keyPEM)
+	if keyDecoded == nil {
+		return nil, fmt.Errorf("CA private key must be PEM-encoded")
+	}
+
+	key, err := x509.ParseECPrivateKey(keyDecoded.Bytes)
```

Comment:
similar question, possible to re-use `tls.DecodePEMKey`?

Reply:
Good :eyes:. Actually, I can use [`tls.ReadPEMCreds()`](https://github.com/linkerd/linkerd2/blob/3d58c9d86e075d6a20675f2c00bac147106c601f/pkg/tls/cred.go#L139) in the webhook, and remove this new method altogether.